zulip/integrations/confluence/confluence.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Confluence integration for Zulip
+#
+# Copyright © 2014 Zulip, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+import zulip
+import sys
+import json
+import requests
+
+from six.moves.configparser import ConfigParser, NoSectionError, NoOptionError
+
+CONFIGFILE = os.path.expanduser("~/.zulip_confluencerc")
+
+INSTRUCTIONS = r"""
+confluence-bot --config-file=~/.zuliprc --search="@nprnews,quantum physics"
+Send Confluence space and page  to a Zulip stream.
+
+To use this script:
+0. Use `Confluence rest api ` to  `Confluence `
+1. Set up Confluence authentication, as described below
+2. Set up a Zulip bot user and download its `.zuliprc`
+   config file to e.g. `~/.zuliprc`
+3. Subscribe the bot to the stream that will receive Confluence updates
+   (default stream: confluence)
+4. Test the script by running it manually, like this:
+5. Configure a crontab entry for this script. A sample crontab entry
+that will process spaces and pages every 5 minutes is:
+*/5 * * * * /usr/local/share/zulip/integrations/confluence/confluence [options]
+
+== Setting up Confluence authentications ==
+
+Run this on a personal or trusted machine, because your API key is
+visible to local users through the command line or config file.
+
+This bot uses OAuth to authenticate with Confluence. Please create a
+~/.zulip_confluencerc with the following contents:
+
+[Confluence_api]
+username =
+consumer_key =
+domain_Url =
+config_file=
```

Comment:
Shouldn't this be local_file since you are using "local_file" to get this value on line 92?

Reply:
I want that user give the file name in  .confluencerc ,so that i can fetch that name or if I use it local what name would I give to  local file?

Reply:
What I meant by that was you had specified the key "config_file" in the instructions to instruct the user to include that key in the zulip_confluencerc but when you were trying to read from zulip_confluencerc, you were trying to get "local_file" key so it should have raised a "NoOptionError".
I'm not sure what would be the proper way to use the local file that you are trying to use(whether to pass it as a commandline argument or something else entirely). @eeshangarg would be able to answer that.